metrics/lib/common.bash

```diff
@@ -92,6 +92,24 @@ check_images()
 	done
 }
 
+# This function performs a docker build on the image names
+# passed in, to ensure that we have the latest changes from
+# the dockerfiles
+check_dockerfiles_images()
+{
+	local image="$1"
+	local dockerfile_path="$2"
+	if [ -z "$image" ] || [ -z "$dockerfile_path" ]; then
+		die "Missing image or dockerfile path variable"
+		exit 1;
+	fi
+	echo "docker building $image"
+	if ! docker build --label "$image" --tag "${image}:latest" -< "$dockerfile_path"; then
```

Comment:
This will build every time I think, which is fine - it is better than only ever building once.
Ideally I'd have liked to have buried a 'version' in the Dockerfile itself and then done a dynamic check here to see if we already had the latest version built and installed or not - which saves us a relatively lengthy docker image build every time. But, I think that is a little tricker and maybe for another day. @GabyCT can you open an Issue noting that we should make the metrics automatic docker image builds smarter to reduce un-necessary rebuilds pls.